data_structures/binary_search_tree/test_bst.py

```diff
@@ -0,0 +1,134 @@
+import pytest
+
+
+def test_insert_into_empty(empty_root):
+    """add single node to empty bst"""
+    empty_root.insert(10)
+    assert empty_root.root.val == 10
+
+
+def test_insert_into_empty_false_value(empty_root):
+    """add a string to empty bst"""
+    with pytest.raises(TypeError):
+        assert empty_root.insert('shoe')
```

Comment:
When you are testing for an exception you do not need an `assert`. The test passes when this line of code raises a `TypeError`.
```python
with pytest.raises(TypeError):
       empty_root.insert('shoe')
```